crypto/src/ed25519.rs

```diff
@@ -341,6 +341,9 @@ impl AggregateAuthenticator for Ed25519AggregateSignature {
         let mut batch = batch::Verifier::new();
 
         for i in 0..pks.len() {
+            if pks[i].len() != sigs[i].0.len() {
```

Comment:
1. Should we apply the same fix to Ed25519PublicKey:: verify_batch() at line 70?
2. Just curious, when `batch` is empty, does the verification succeed or fail?

Reply:
Great points we should address both. Re 2nd: **empty batch should be considered a fail**; by experience we cannot accepting empty crypto materials as it could lead to surprises.